src/diffusers/loaders/ip_adapter.py

```diff
@@ -155,3 +157,34 @@ def set_ip_adapter_scale(self, scale):
         for attn_processor in self.unet.attn_processors.values():
             if isinstance(attn_processor, (IPAdapterAttnProcessor, IPAdapterAttnProcessor2_0)):
                 attn_processor.scale = scale
+
+    def unload_ip_adapter(self):
+        """
+        Unloads the IP Adapter weights
+
+        Examples:
+
+        ```python
+        >>> # Assuming `pipeline` is already loaded with the IP Adapter weights.
+        >>> pipeline.unload_ip_adapter()
+        >>> ...
+        ```
+        """
+        # remove CLIP image encoder
+        if hasattr(self, "image_encoder"):
+            self.image_encoder = None
+            self.register_to_config(image_encoder=[None, None])
+
+        # remove feature extractor
+        if hasattr(self, "feature_extractor"):
+            self.feature_extractor = None
+            self.register_to_config(feature_extractor=[None, None])
+
+        # remove hidden encoder
+        self.unet.encoder_hid_proj = None
+        self.config.encoder_hid_dim_type = None
+
+        # restore original Unet layers
+        device = self.unet.device
```

Comment:
I was thinking, we just have to set the attention processor back to default here instead of reloading the UNet again.

cc @sayakpaul here, let me know what you think!

Reply:
Indeed that should be the way to go. We can first store the original attention processors and wap them back in during the unloading process. 

Here's a reference that might be relevant: 

https://github.com/huggingface/diffusers/blob/d704a730cd6d2961d9ec17d7a041c4e9f7e13812/src/diffusers/models/unet_2d_condition.py#L831